ReflectionUtils: java.lang.AssertionError: assertion failed: unsafe symbol ... (child of <none>) in runtime reflection universe

```
java.lang.AssertionError: assertion failed: unsafe symbol sqlContext (child of <none>) in runtime reflection universe
	at scala.reflect.internal.Symbols$Symbol.<init>(Symbols.scala:205)
	at scala.reflect.internal.Symbols$TermSymbol.<init>(Symbols.scala:2768)
	at scala.reflect.internal.Symbols$StubTermSymbol.<init>(Symbols.scala:3524)
	at scala.reflect.internal.Symbols$class.newStubSymbol(Symbols.scala:192)
	at scala.reflect.internal.SymbolTable.newStubSymbol(SymbolTable.scala:16)
	at scala.reflect.internal.Symbols$Symbol.newStubSymbol(Symbols.scala:521)
	at scala.reflect.internal.pickling.UnPickler$Scan.readExtSymbol$1(UnPickler.scala:258)
	at scala.reflect.internal.pickling.UnPickler$Scan.readSymbol(UnPickler.scala:286)
	at scala.reflect.internal.pickling.UnPickler$Scan$LazyTypeRefAndAlias$$anonfun$8.apply(UnPickler.scala:774)
	at scala.reflect.internal.pickling.UnPickler$Scan$LazyTypeRefAndAlias$$anonfun$8.apply(UnPickler.scala:774)
	at scala.reflect.internal.pickling.UnPickler$Scan.at(UnPickler.scala:179)
	at scala.reflect.internal.pickling.UnPickler$Scan$LazyTypeRefAndAlias.completeInternal(UnPickler.scala:774)
	at scala.reflect.internal.pickling.UnPickler$Scan$LazyTypeRef.complete(UnPickler.scala:761)
	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1535)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$15.scala$reflect$runtime$SynchronizedSymbols$SynchronizedSymbol$$super$info(SynchronizedSymbols.scala:198)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anonfun$info$1.apply(SynchronizedSymbols.scala:127)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anonfun$info$1.apply(SynchronizedSymbols.scala:127)
	at scala.reflect.runtime.Gil$class.gilSynchronized(Gil.scala:19)
	at scala.reflect.runtime.JavaUniverse.gilSynchronized(JavaUniverse.scala:16)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$class.gilSynchronizedIfNotThreadsafe(SynchronizedSymbols.scala:123)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$15.gilSynchronizedIfNotThreadsafe(SynchronizedSymbols.scala:198)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$class.info(SynchronizedSymbols.scala:127)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$15.info(SynchronizedSymbols.scala:198)
	at scala.reflect.internal.Symbols$Symbol.initialize(Symbols.scala:1680)
	at scala.reflect.internal.Symbols$Symbol.annotations(Symbols.scala:1847)
	at scala.reflect.internal.Symbols$Symbol.addAnnotation(Symbols.scala:1866)
	at scala.reflect.internal.pickling.UnPickler$Scan.readSymbolAnnotation(UnPickler.scala:517)
	at scala.reflect.internal.pickling.UnPickler$Scan.run(UnPickler.scala:97)
	at scala.reflect.internal.pickling.UnPickler.unpickle(UnPickler.scala:38)
	at scala.reflect.runtime.JavaMirrors$JavaMirror.unpickleClass(JavaMirrors.scala:619)
	at scala.reflect.runtime.SymbolLoaders$TopClassCompleter$$anonfun$complete$1.apply$mcV$sp(SymbolLoaders.scala:28)
	at scala.reflect.runtime.SymbolLoaders$TopClassCompleter$$anonfun$complete$1.apply(SymbolLoaders.scala:25)
	at scala.reflect.runtime.SymbolLoaders$TopClassCompleter$$anonfun$complete$1.apply(SymbolLoaders.scala:25)
	at scala.reflect.internal.SymbolTable.slowButSafeEnteringPhaseNotLaterThan(SymbolTable.scala:263)
	at scala.reflect.runtime.SymbolLoaders$TopClassCompleter.complete(SymbolLoaders.scala:25)
	at scala.reflect.internal.Symbols$Symbol.info(Symbols.scala:1535)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$8.scala$reflect$runtime$SynchronizedSymbols$SynchronizedSymbol$$super$info(SynchronizedSymbols.scala:168)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anonfun$info$1.apply(SynchronizedSymbols.scala:127)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anonfun$info$1.apply(SynchronizedSymbols.scala:127)
	at scala.reflect.runtime.Gil$class.gilSynchronized(Gil.scala:19)
	at scala.reflect.runtime.JavaUniverse.gilSynchronized(JavaUniverse.scala:16)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$class.gilSynchronizedIfNotThreadsafe(SynchronizedSymbols.scala:123)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$8.gilSynchronizedIfNotThreadsafe(SynchronizedSymbols.scala:168)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$class.info(SynchronizedSymbols.scala:127)
	at scala.reflect.runtime.SynchronizedSymbols$SynchronizedSymbol$$anon$8.info(SynchronizedSymbols.scala:168)
	at scala.reflect.internal.Types$TypeRef.decls(Types.scala:2224)
	at scala.reflect.internal.Types$TypeRef.decls(Types.scala:2129)
	at za.co.absa.commons.reflect.ReflectionUtils$$anonfun$1.apply(ReflectionUtils.scala:123)
	at za.co.absa.commons.reflect.ReflectionUtils$$anonfun$1.apply(ReflectionUtils.scala:122)
	at scala.util.Try$.apply(Try.scala:192)
	at za.co.absa.commons.reflect.ReflectionUtils$.reflectClass$1(ReflectionUtils.scala:122)
	at za.co.absa.commons.reflect.ReflectionUtils$.reflectClassHierarchy$1(ReflectionUtils.scala:112)
	at za.co.absa.commons.reflect.ReflectionUtils$.extractFieldValue(ReflectionUtils.scala:168)
	at za.co.absa.commons.reflect.ReflectionUtils$.extractFieldValue(ReflectionUtils.scala:183)
	at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$$anonfun$baseRelationProcessor$1.applyOrElse(BigQueryPlugin.scala:53)
	at za.co.absa.spline.harvester.plugin.embedded.BigQueryPlugin$$anonfun$baseRelationProcessor$1.applyOrElse(BigQueryPlugin.scala:50)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at za.co.absa.spline.harvester.plugin.composite.LogicalRelationPlugin$$anonfun$1.applyOrElse(LogicalRelationPlugin.scala:37)
	at za.co.absa.spline.harvester.plugin.composite.LogicalRelationPlugin$$anonfun$1.applyOrElse(LogicalRelationPlugin.scala:34)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at za.co.absa.spline.harvester.plugin.embedded.SQLPlugin$$anonfun$1.applyOrElse(SQLPlugin.scala:47)
	at za.co.absa.spline.harvester.plugin.embedded.SQLPlugin$$anonfun$1.applyOrElse(SQLPlugin.scala:47)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at za.co.absa.spline.harvester.plugin.embedded.DataSourceV2Plugin$$anonfun$1.applyOrElse(DataSourceV2Plugin.scala:38)
	at za.co.absa.spline.harvester.plugin.embedded.DataSourceV2Plugin$$anonfun$1.applyOrElse(DataSourceV2Plugin.scala:38)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at za.co.absa.spline.harvester.builder.read.PluggableReadCommandExtractor$$anonfun$asReadCommand$1.applyOrElse(PluggableReadCommandExtractor.scala:41)
	at za.co.absa.spline.harvester.builder.read.PluggableReadCommandExtractor$$anonfun$asReadCommand$1.applyOrElse(PluggableReadCommandExtractor.scala:38)
	at scala.PartialFunction$Lifted.apply(PartialFunction.scala:223)
	at scala.PartialFunction$Lifted.apply(PartialFunction.scala:219)
	at scala.PartialFunction$.condOpt(PartialFunction.scala:286)
	at za.co.absa.spline.harvester.builder.read.PluggableReadCommandExtractor.asReadCommand(PluggableReadCommandExtractor.scala:38)
	at za.co.absa.spline.harvester.LineageHarvester.za$co$absa$spline$harvester$LineageHarvester$$createOperationBuilder(LineageHarvester.scala:193)
	at za.co.absa.spline.harvester.LineageHarvester$$anonfun$13.apply(LineageHarvester.scala:169)
	at za.co.absa.spline.harvester.LineageHarvester$$anonfun$13.apply(LineageHarvester.scala:169)
	at scala.Option.getOrElse(Option.scala:121)
	at za.co.absa.spline.harvester.LineageHarvester.traverseAndCollect$1(LineageHarvester.scala:169)
	at za.co.absa.spline.harvester.LineageHarvester.za$co$absa$spline$harvester$LineageHarvester$$createOperationBuildersRecursively(LineageHarvester.scala:188)
	at za.co.absa.spline.harvester.LineageHarvester$$anonfun$harvest$2.apply(LineageHarvester.scala:66)
	at za.co.absa.spline.harvester.LineageHarvester$$anonfun$harvest$2.apply(LineageHarvester.scala:64)
	at scala.Option.flatMap(Option.scala:171)
	at za.co.absa.spline.harvester.LineageHarvester.harvest(LineageHarvester.scala:64)
	at za.co.absa.spline.harvester.QueryExecutionEventHandler.onSuccess(QueryExecutionEventHandler.scala:42)
	at za.co.absa.spline.harvester.listener.SplineQueryExecutionListener$$anonfun$onSuccess$1$$anonfun$apply$mcV$sp$1.apply(SplineQueryExecutionListener.scala:40)
	at za.co.absa.spline.harvester.listener.SplineQueryExecutionListener$$anonfun$onSuccess$1$$anonfun$apply$mcV$sp$1.apply(SplineQueryExecutionListener.scala:40)
	at scala.Option.foreach(Option.scala:257)
	at za.co.absa.spline.harvester.listener.SplineQueryExecutionListener$$anonfun$onSuccess$1.apply$mcV$sp(SplineQueryExecutionListener.scala:40)
	at za.co.absa.spline.harvester.listener.SplineQueryExecutionListener.withErrorHandling(SplineQueryExecutionListener.scala:49)
	at za.co.absa.spline.harvester.listener.SplineQueryExecutionListener.onSuccess(SplineQueryExecutionListener.scala:39)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1$$anonfun$apply$mcV$sp$1.apply(QueryExecutionListener.scala:124)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1$$anonfun$apply$mcV$sp$1.apply(QueryExecutionListener.scala:123)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$org$apache$spark$sql$util$ExecutionListenerManager$$withErrorHandling$1.apply(QueryExecutionListener.scala:145)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$org$apache$spark$sql$util$ExecutionListenerManager$$withErrorHandling$1.apply(QueryExecutionListener.scala:143)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
	at scala.collection.mutable.ListBuffer.foreach(ListBuffer.scala:45)
	at org.apache.spark.sql.util.ExecutionListenerManager.org$apache$spark$sql$util$ExecutionListenerManager$$withErrorHandling(QueryExecutionListener.scala:143)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1.apply$mcV$sp(QueryExecutionListener.scala:123)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1.apply(QueryExecutionListener.scala:123)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1.apply(QueryExecutionListener.scala:123)
	at org.apache.spark.sql.util.ExecutionListenerManager.readLock(QueryExecutionListener.scala:156)
	at org.apache.spark.sql.util.ExecutionListenerManager.onSuccess(QueryExecutionListener.scala:122)
	at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:678)
	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:285)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:271)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:229)
	at com.google.cloud.spark.bigquery.BigQueryWriteHelper.writeDataFrameToBigQuery(BigQueryWriteHelper.scala:88)
	at com.google.cloud.spark.bigquery.BigQueryInsertableRelation.insert(BigQueryInsertableRelation.scala:43)
	at com.google.cloud.spark.bigquery.BigQueryRelationProvider.createRelation(BigQueryRelationProvider.scala:113)
	at org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand.run(SaveIntoDataSourceCommand.scala:45)
	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult$lzycompute(commands.scala:70)
	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult(commands.scala:68)
	at org.apache.spark.sql.execution.command.ExecutedCommandExec.doExecute(commands.scala:86)
	at org.apache.spark.sql.execution.SparkPlan$$anonfun$execute$1.apply(SparkPlan.scala:131)
	at org.apache.spark.sql.execution.SparkPlan$$anonfun$execute$1.apply(SparkPlan.scala:127)
	at org.apache.spark.sql.execution.SparkPlan$$anonfun$executeQuery$1.apply(SparkPlan.scala:155)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:152)
	at org.apache.spark.sql.execution.SparkPlan.execute(SparkPlan.scala:127)
	at org.apache.spark.sql.execution.QueryExecution.toRdd$lzycompute(QueryExecution.scala:80)
	at org.apache.spark.sql.execution.QueryExecution.toRdd(QueryExecution.scala:80)
	at org.apache.spark.sql.DataFrameWriter$$anonfun$runCommand$1.apply(DataFrameWriter.scala:676)
	at org.apache.spark.sql.DataFrameWriter$$anonfun$runCommand$1.apply(DataFrameWriter.scala:676)
	at org.apache.spark.sql.execution.SQLExecution$$anonfun$withNewExecutionId$1.apply(SQLExecution.scala:78)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:125)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:73)
	at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:676)
	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:285)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:271)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:229)
	at za.co.absa.spline.example.bigquery.BigQueryExample$.delayedEndpoint$za$co$absa$spline$example$bigquery$BigQueryExample$1(BigQueryExample.scala:43)
	at za.co.absa.spline.example.bigquery.BigQueryExample$delayedInit$body.apply(BigQueryExample.scala:22)
	at scala.Function0$class.apply$mcV$sp(Function0.scala:34)
	at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:12)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.App$$anonfun$main$1.apply(App.scala:76)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
	at scala.App$class.main(App.scala:76)
	at za.co.absa.spline.SparkApp.main(SparkApp.scala:27)
```

--- a/src/main/scala/za/co/absa/commons/reflect/ReflectionUtils.scala
+++ b/src/main/scala/za/co/absa/commons/reflect/ReflectionUtils.scala
@@ -19,7 +19,6 @@ package za.co.absa.commons.reflect
 import scala.annotation.tailrec
 import scala.collection.concurrent.TrieMap
 import scala.reflect.ClassTag
-import scala.reflect.internal.Symbols
 import scala.reflect.runtime.universe._
 import scala.tools.reflect.ToolBox
 
@@ -120,26 +119,23 @@ object ReflectionUtils {
       }
 
     def reflectClass(c: Class[_]) = {
-      val members =
-        try mirror.classSymbol(c).toType.decls
-        catch {
-          // a workaround for Scala bug #12190
-          case _: Symbols#CyclicReference => Nil
-          // error reading Scala signature
-          case e: RuntimeException
-            if e.getMessage.toLowerCase.contains("scala signature") => Nil
-        }
+      val maybeMember = util.Try {
+        val members = mirror.classSymbol(c).toType.decls
+        members
+          .filter(smb => (
+            smb.toString.endsWith(s" $fieldName")
+              && smb.isTerm
+              && !smb.isConstructor
+              && (!smb.isMethod || smb.asMethod.paramLists.forall(_.isEmpty))
+            ))
+          .minBy(!_.isMethod)
 
-      val maybeMember = members
-        .filter(smb => (
-          smb.toString.endsWith(s" $fieldName")
-            && smb.isTerm
-            && !smb.isConstructor
-            && (!smb.isMethod || smb.asMethod.paramLists.forall(_.isEmpty))
-          ))
-        .toArray
-        .sortBy(!_.isMethod) // method members first
-        .headOption
+        // Unless:
+        //   - `Symbols#CyclicReference` (a workaround for Scala bug #12190)
+        //   - `RuntimeException("scala signature")` (#80, #82)
+        //   - ... or any other runtime error during Scala reflection
+        // }
+      }.toOption
 
       maybeMember
         .map(m => {

